### DISCUSSION TITLE
useQuery vs useInfiniteQuery on mount/unmount

### DISCUSSION THREAD

**Discussion by tamis-laan** [TanStack/query#3677] [Q&A]
I'm using a `useQuery` and `useInfiniteQuery` in two different components. The components are mounted and unmounted when navigating to and when navigating away from a screen that include these components. 

The `useQuery` does not refetch the data on unmount->mount while the `useInfiniteQuery` does refetch the data. Both queries use a stale time of `staleTime: 20*60*1000` 20 minutes.

Why does this behavior differ??

**Comment by TkDodo** [comment]
It shouldn't be different. Please show a codesandbox reproduction
